sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcsOptions.java

```diff
@@ -48,20 +48,20 @@ public interface GcsOptions extends ApplicationNameOptions, GcpOptions, Pipeline
 
   /**
    * The ExecutorService instance to use to create threads, can be overridden to specify an
-   * ExecutorService that is compatible with the user's environment. If unset, the default is to
-   * create an ExecutorService with an unbounded number of threads; this is compatible with Google
-   * AppEngine.
+   * ExecutorService that is compatible with the user's environment. If unset, the default is to use
+   * {@link ExecutorOptions#getScheduledExecutorService()}.
+   *
+   * @deprecated use {@link ExecutorOptions#getScheduledExecutorService()} instead
    */
   @JsonIgnore
-  @Description(
-      "The ExecutorService instance to use to create multiple threads. Can be overridden "
-          + "to specify an ExecutorService that is compatible with the user's environment. If unset, "
-          + "the default is to create an ExecutorService with an unbounded number of threads; this "
-          + "is compatible with Google AppEngine.")
   @Default.InstanceFactory(ExecutorServiceFactory.class)
   @Hidden
+  @Deprecated
   ExecutorService getExecutorService();
 
+  /** @deprecated use {@link ExecutorOptions#setScheduledExecutorService} instead. If set, it may
+   * result in multiple ExecutorServices, and therefor thread pools, in the runtime  */
```

Comment:
```suggestion
  /**
   * @deprecated use {@link ExecutorOptions#setScheduledExecutorService} instead. If set, it may
   *     result in multiple ExecutorServices, and therefore thread pools, in the runtime.
   */
```